### ISSUE TITLE
Issue cloning HTTP headers?

### ISSUE THREAD

**Issue by maiorfi** [node-red/node-red#97]
I set up a simple http-in (put) request/response node couple, but when I send my flow a request via "Dev HTTP Client" chrome app, I get the following response:

TypeError: Cannot set property headerSent of #<ServerResponse> which has only a getter
    at _clone (C:\Users\lorenzo\Downloads\node-red\node_modules\clone\clone.js:92:20)
    at _clone (C:\Users\lorenzo\Downloads\node-red\node_modules\clone\clone.js:92:22)
    at _clone (C:\Users\lorenzo\Downloads\node-red\node_modules\clone\clone.js:92:22)
    at _clone (C:\Users\lorenzo\Downloads\node-red\node_modules\clone\clone.js:92:22)
    at clone (C:\Users\lorenzo\Downloads\node-red\node_modules\clone\clone.js:105:18)
    at HTTPIn.Node.send (C:\Users\lorenzo\Downloads\node-red\red\nodes.js:174:41)
    at HTTPIn.callback (C:\Users\lorenzo\Downloads\node-red\nodes\core\io\21-httpin.js:35:19)
    at callbacks (C:\Users\lorenzo\Downloads\node-red\node_modules\express\lib\router\index.js:164:37)
    at urlencoded (C:\Users\lorenzo\Downloads\node-red\node_modules\express\node_modules\connect\lib\middleware\urlencoded.js:46:27)
    at callbacks (C:\Users\lorenzo\Downloads\node-red\node_modules\express\lib\router\index.js:164:37)

**Comment by btsimonh** [comment]
@knolleary - may have returned 0.17.5 - I'm getting:
"TypeError: Cannot set property listening of #<Server> which has only a getter".
removing a wire from http input to a debug node solved it.
error was reported against the following function node, where I am doing newmsg.originalmsg = msg, but that should create a ref not a clone...? (the next step is an http request, and only after that the response node where I restore the original with newmsg = msg.originalmsg).

**Comment by btsimonh** [comment]
raising a new issue.
